plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergV2.java

```diff
@@ -706,6 +726,75 @@ public void testDeletingEntireFileWithNonTupleDomainConstraint()
+        assertQuery(format("SELECT COUNT(*) from %s WHERE event_date = DATE '2022-11-01' AND country IN ('USA', 'IND')", tableName), "SELECT 4");
+
+        assertQuery(clientSession, format("SELECT COUNT(*) from %s WHERE event_date >= DATE '2022-11-01' and country = 'USA'", tableName), "SELECT 4");
+        assertQuery(format("SELECT COUNT(*) from %s WHERE event_date >= DATE '2022-11-01' and country = 'USA'", tableName), "SELECT 4");
+
+        assertUpdate(format("DELETE FROM %s WHERE event_date = DATE '2022-11-01' AND country ='USA' AND state = 'California'", tableName), 1);
+
+        assertQuery(clientSession, format("SELECT COUNT(*) from %s WHERE event_date = DATE '2022-11-01'", tableName), "SELECT 3");
+        assertQuery(format("SELECT COUNT(*) from %s WHERE event_date = DATE '2022-11-01'", tableName), "SELECT 3");
+
+        assertQuery(clientSession, format("SELECT COUNT(*) from %s WHERE event_date = DATE '2022-11-01' AND state = 'California'", tableName), "SELECT 0");
+        assertQuery(format("SELECT COUNT(*) from %s WHERE event_date = DATE '2022-11-01' AND state = 'California'", tableName), "SELECT 0");
+
+        assertQuery(clientSession, format("SELECT COUNT(*) from %s WHERE event_date = DATE '2022-11-02'", tableName), "SELECT 2");
+        assertQuery(format("SELECT COUNT(*) from %s WHERE event_date = DATE '2022-11-02'", tableName), "SELECT 2");
+
+        assertQuery(clientSession, format("SELECT COUNT(*) from %s WHERE event_date = DATE '2022-11-04'", tableName), "SELECT 1");
+        assertQuery(format("SELECT COUNT(*) from %s WHERE event_date = DATE '2022-11-04'", tableName), "SELECT 1");
+
+        assertUpdate(format("DELETE FROM %s WHERE event_date = DATE '2022-11-01' AND country ='USA'", tableName), 1);
+
+        assertQuery(clientSession, format("SELECT COUNT(*) from %s WHERE event_date = DATE '2022-11-01'", tableName), "SELECT 2");
+        assertQuery(format("SELECT COUNT(*) from %s WHERE event_date = DATE '2022-11-01'", tableName), "SELECT 2");
+
+        assertQuery(clientSession, format("SELECT COUNT(*) from %s WHERE event_date = DATE '2022-11-01'", tableName), "SELECT 2");
+        assertQuery(format("SELECT COUNT(*) from %s WHERE event_date = DATE '2022-11-01'", tableName), "SELECT 2");
+
+        assertQuery(clientSession, format("SELECT COUNT(*) from %s WHERE event_date = DATE '2022-11-02' AND country = 'FRA' AND state = 'Brittany'", tableName), "SELECT 1");
+        assertQuery(clientSession, format("SELECT COUNT(*) from %s WHERE event_date = DATE '2022-11-02' AND country = 'FRA' AND state = 'Brittany'", tableName), "SELECT 1");
+
+        assertQuery(clientSession, format("SELECT COUNT(*) from %s WHERE event_date = DATE '2022-11-02' AND country = 'USA' AND state = 'Corsica'", tableName), "SELECT 1");
+        assertQuery(format("SELECT COUNT(*) from %s WHERE event_date = DATE '2022-11-02' AND country = 'USA' AND state = 'Corsica'", tableName), "SELECT 1");
```

Comment:
I think we need some tests which validate that agg pushdown actually happened by looking at the explain plan?